lib/tasks/csv.rake

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require "csv"
+
+=begin
+
+These tasks are designed to be used when employing a spreadsheet of
+metadata to import ephemera materials into Figgy.
+
+=end
+
+namespace :csv do
+
+  desc "ingest ephemera from a csv file"
+  task ingest_ephemera: :environment do
+    basedir = ENV["BASEDIR"]
+    input_project_id = ENV["PROJECT"]
+    csvfile = ENV["CSV"]
+
+    abort "usage: PROJECT=project_id BASEDIR=directory CSV=csvfile rake csv:ingest_ephemera" unless input_project_id && basedir
+    abort "no such file #{csvfile}" unless File.file?(csvfile)
+    class_name = "ScannedResource"
+    @logger = Logger.new(STDOUT)
+
+    qs = Valkyrie::MetadataAdapter.find(:indexing_persister).query_service
+    project = qs.find_by(id: input_project_id)
+
+    abort "no such project" unless project
+
+    change_set_persister = ChangeSetPersister.new(
+      metadata_adapter: Valkyrie::MetadataAdapter.find(:indexing_persister),
+      storage_adapter: Valkyrie::StorageAdapter.find(:disk_via_copy))
+
+    @logger.info "beginning to ingest ephemera into #{project.title} from #{csvfile} with basedir #{basedir}"
+    IngestEphemeraCSVJob.perform_now(project.id, csvfile, basedir)
+    @logger.info "finished ingesting ephemera"
+  end
+
+  desc "Ingest resources from a csv file"
+  task from_csv: :environment do
```

Comment:
This task will be invoked as `csv:from_csv` -- can it be named something more descriptive? Like with `csv:ingest_ephemera` above it is easier to see the intention.

Reply:
good point; changed the name to ingest_resources in the latest commit.